### PR TITLE
angular: implement formService

### PR DESCRIPTION
**Pull request by mshima**
Closes https://github.com/jhipster/generator-jhipster/issues/10198

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by DanielFran — APPROVED** [review]

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  let service: <%= entityAngularName %>FormService;
+  let elemDefault: I<%= entityAngularName %>;
+<%_ if (fields.some(field => field.fieldTypeTimed)) { _%>
+  let currentTime: dayjs.Dayjs;
+<%_ } _%>
+<%_ if (fields.some(field => field.fieldTypeLocalDate)) { _%>
+  let currentDate: dayjs.Dayjs;
+<%_ } _%>
+
+
```
Comment (qmonmert) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs:45]: remove one line break

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  <%_ } else if (field.fieldTypeCharSequence) { _%>
+      <%= field.fieldName %>: 'AAAAAAA',
+  <%_ } else if (field.fieldTypeLocalDate) { _%>
+      <%= field.fieldName %>: currentDate,
+  <%_ } else if (field.fieldTypeTimed) { _%>
+      <%= field.fieldName %>: currentTime,
+  <%_ } else if (field.fieldTypeBinary && !field.blobContentTypeText) { _%>
+      <%= field.fieldName %>ContentType: 'image/png',
+      <%= field.fieldName %>: 'AAAAAAA',
+  <%_ } else if (!field.fieldIsEnum) { /* (fieldType === 'byte[]' || fieldType === 'ByteBuffer') && fieldTypeBlobContent === 'any' || (fieldType === 'byte[]' || fieldType === 'ByteBuffer') && fieldTypeBlobContent === 'image' || fieldType === 'LocalDate' */ _%>
```
Comment (qmonmert) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs:77]: I don't understand this comment?

**Review by qmonmert — DISMISSED** [review]
LGTM (with my 2 comments)

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+describe('<%= entityAngularName %> Form Service', () => {
+  let service: <%= entityAngularName %>FormService;
+  let elemDefault: I<%= entityAngularName %>;
+<%_ if (fields.some(field => field.fieldTypeTimed)) { _%>
+  let currentTime: dayjs.Dayjs;
+<%_ } _%>
+<%_ if (fields.some(field => field.fieldTypeLocalDate)) { _%>
+  let currentDate: dayjs.Dayjs;
+<%_ } _%>
+
```
Comment (mshima) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs:44]:
```suggestion
```

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.ts.ejs [hunk 4]
```diff
@@ -0,0 +1,214 @@
+<%#
```
Comment (Tcharl) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.ts.ejs:1]: dumb question: could it be called entity.form.ts?

Comment (mshima) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.ts.ejs:1]: In my opinion makes more sense to call it entity-form.

**Review by DanielFran — APPROVED** [review]

**Comment by mshima** [comment]
@DanielFran bounty claimed https://opencollective.com/generator-jhipster/expenses/90549.

**Comment by DanielFran** [comment]
@mshima approved

**Pull request merged**